File: algorithms/1_easy/F/01979_find_greatest_common_divisor_of_array/solution.c
#define MAX(a, b) ((a) > (b) ? (a) : (b))
#define MIN(a, b) ((a) < (b) ? (a) : (b))

static inline int gcd(int a, int b)
{
    while (b != 0) {
        const int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int findGCD(const int* nums, const int len)
{
    assert(len > 0 && nums != NULL);
    int min = INT_MAX;
    int max = INT_MIN;

    for (int i = 0; i < len; i += 1) {
        min = MIN(min, nums[i]);
        max = MAX(max, nums[i]);
    }

    return gcd(max, min);
}

File: algorithms/1_easy/F/01979_find_greatest_common_divisor_of_array/solution.rs
impl Solution {
    #[inline(always)]
    fn gcd(mut a: i32, mut b: i32) -> i32 {
        while b != 0 {
            let temp = b;
            b = a % b;
            a = temp;
        }
        a
    }

    pub fn find_gcd(nums: Vec<i32>) -> i32 {
        let mut min = i32::MAX;
        let mut max = i32::MIN;

        for num in nums.iter() {
            min = min.min(*num);
            max = max.max(*num);
        }

        Self::gcd(max, min)
    }
}
